### PR TITLE
Add Subworkflow Build_Indices for switch to DSL2

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
Add subworklow for building indices

# nf-core/sarek pull request

Many thanks for contributing to nf-core/sarek!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/sarek branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/sarek)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by ggabernet** [comment]
The simple test passes now!

**Comment by ggabernet** [comment]
yay! test `test` passes again :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [dsl2](https://github.com/nf-core/sarek/tree/dev)
+
+- [#238](https://github.com/nf-core/sarek/pull/238) -Add subworkflow for building all the indices
+
 ## [dev](https://github.com/nf-core/sarek/tree/dev)
 
 - [#234](https://github.com/nf-core/sarek/pull/234) -Switching to DSL2
```
Comment (MaxUlysse) [inline CHANGELOG.md:13]: Maybe move this line to DSL2 too, no?

Comment (ggabernet) [reply CHANGELOG.md:13]: will do in the next PR coming :)

**Pull request merged**